src/shared/bootspec.c

```diff
@@ -1340,14 +1340,19 @@ int show_boot_entry(
         if (e->id)
                 printf("           id: %s\n", e->id);
         if (e->path) {
-                _cleanup_free_ char *link = NULL;
+                _cleanup_free_ char *text = NULL, *link = NULL;
+
+                const char *p = e->root ? path_startswith(e->path, e->root) : NULL;
+                if (p &&
+                    asprintf(&text, "%s%s/%s/%s", ansi_grey(), e->root, ansi_normal(), p) < 0)
```

Comment:
so far we had the rule of avoiding asprintf() if you just want to concat strings. use strjoin() instead. a million times faster...

Reply:
> so far we had the rule of avoiding asprintf() if you just want to concat strings. use strjoin() instead. a million times faster...

Updated.